sdk/core/Azure.Core/api/Azure.Core.net461.cs

```diff
@@ -472,6 +532,58 @@ public partial class RequestUriBuilder
         public override string ToString() { throw null; }
         public System.Uri ToUri() { throw null; }
     }
+    public sealed partial class ResourceIdentifier : System.IComparable<Azure.Core.ResourceIdentifier>, System.IEquatable<Azure.Core.ResourceIdentifier>
+    {
+        public static readonly Azure.Core.ResourceIdentifier Root;
+        public ResourceIdentifier(string resourceId) { }
+        public Azure.Core.AzureLocation? Location { get { throw null; } }
+        public string Name { get { throw null; } }
+        public Azure.Core.ResourceIdentifier Parent { get { throw null; } }
+        public string? Provider { get { throw null; } }
+        public string? ResourceGroupName { get { throw null; } }
+        public Azure.Core.ResourceType ResourceType { get { throw null; } }
+        public string? SubscriptionId { get { throw null; } }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public Azure.Core.ResourceIdentifier AppendChildResource(string childResourceType, string childResourceName) { throw null; }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public Azure.Core.ResourceIdentifier AppendProviderResource(string providerNamespace, string resourceType, string resourceName) { throw null; }
+        public int CompareTo(Azure.Core.ResourceIdentifier? other) { throw null; }
+        public bool Equals(Azure.Core.ResourceIdentifier? other) { throw null; }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override bool Equals(object? obj) { throw null; }
+        [System.ComponentModel.EditorBrowsableAttribute(System.ComponentModel.EditorBrowsableState.Never)]
+        public override int GetHashCode() { throw null; }
+        public static bool operator ==(Azure.Core.ResourceIdentifier left, Azure.Core.ResourceIdentifier right) { throw null; }
+        public static bool operator >(Azure.Core.ResourceIdentifier left, Azure.Core.ResourceIdentifier right) { throw null; }
```

Comment:
We might want to remove IComparable implementation the same way we did for the location

Reply:
This supports sorting for large lists given the structure of having subscription first followed by resource group this is useful when processing sets of resource ids from a data source

Reply:
Someone can implement comparer like that if they need it.

Reply:
yes they can, we discussed this previously and landed on the side of leaving it in for convenience since its a fairly common usecase

Reply:
btw, You don't need operators for sorting. Only CompareTo

Reply:
If you don't implement IComparable List<T>.Sort() won't work it will make you pass in an IComparer.  If we implement IComparable it makes you implement the operators.  In our last discussion we landed on leaving it in so that sorting will be easier for customers.